packages/core/src/browser/dialogs/react-dialog.tsx

```diff
@@ -0,0 +1,58 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as ReactDOM from 'react-dom';
+import * as React from 'react';
+import { injectable, inject } from 'inversify';
+import { DisposableCollection, Disposable } from '../../common';
+import { Message } from '../widgets';
+import { AbstractDialog } from './abstract-dialog';
+import { DialogProps } from './dialogs';
+
+// Modified version of ReactWidget, based on AbstractDialog
+
+@injectable()
+export abstract class ReactDialog extends AbstractDialog<void> {
+    protected readonly onRender = new DisposableCollection();
+
+    constructor(
+        @inject(DialogProps) protected readonly props: DialogProps
+    ) {
+        super(props);
+        this.scrollOptions = {
+            suppressScrollX: true,
+            minScrollbarLength: 35,
+        };
```

Comment:
It's not clear why this should be here.  scrollOptions is not React-specific, so surely should either be in AbstractDialog so it applies to all dialogs or it should be removed from here and only put in final classes that need it.

Reply:
I just took it from the react widget, not sure what it does :)

Reply:
Heh, I actually have no idea what that does. I just copied it over from ReactWidget. I have no idea why that was done there, though... Maybe the scrollbar was glitching out when react re-rendered or something?